bundle/jsky.app.ot/src/main/java/jsky/app/ot/gemini/editor/targetComponent/EdCompTargetList.java

```diff
@@ -1079,20 +1029,12 @@ protected HorizonsReply doInBackgroundWithSubject() {
 
                 service.setObjectType(null);
 
-                //if the object is an Id, set the appropriate type in
-                //the query, so we can attempt to narrow it.
-                //UPDATE: why set it only for IDs? Now we'll always set it.
-                //  if (_nameIsId(name)) {
-                NonSiderealTargetSupport.NonSiderealSystem system =
-                        (NonSiderealTargetSupport.NonSiderealSystem) _w.orbitalElementFormat.getSelectedItem();
-                if (system == _nonSiderealTargetSup.getNonSiderealSystem(ITarget.Tag.JPL_MINOR_BODY)) {
-                    service.setObjectType(HorizonsQuery.ObjectType.COMET);
-                } else if (system == _nonSiderealTargetSup.getNonSiderealSystem(ITarget.Tag.MPC_MINOR_PLANET)) {
-                    service.setObjectType(HorizonsQuery.ObjectType.MINOR_BODY);
-                } else if (system == _nonSiderealTargetSup.getNonSiderealSystem(ITarget.Tag.NAMED)) {
-                    service.setObjectType(HorizonsQuery.ObjectType.MAJOR_BODY);
+                ITarget.Tag tag = (ITarget.Tag) _w.orbitalElementFormat.getSelectedItem();
+                switch (tag) {
+                    case JPL_MINOR_BODY:   service.setObjectType(HorizonsQuery.ObjectType.COMET);
+                    case MPC_MINOR_PLANET: service.setObjectType(HorizonsQuery.ObjectType.MINOR_BODY);
+                    case NAMED:            service.setObjectType(HorizonsQuery.ObjectType.MAJOR_BODY);
```

Comment:
With Java switch statements on enums I like having a default case that throws an exception.